creator/html_paragraph.go

```diff
@@ -0,0 +1,625 @@
+/*
+ * This file is subject to the terms and conditions defined in
+ * file 'LICENSE.md', which is part of this source code package.
+ */
+
+package creator
+
+import (
+	"bytes"
+	"log"
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/unidoc/unipdf/v3/common"
+	"github.com/unidoc/unipdf/v3/contentstream/draw"
+	"github.com/unidoc/unipdf/v3/model"
+	"github.com/vanng822/css"
+	"golang.org/x/net/html"
+)
+
+// htmlElementStyle is one element only style
+type htmlElementStyle struct {
+	// block style
+
+	// Background
+	backgroundColor *model.PdfColorDeviceRGB
+
+	borderLineStyle draw.LineStyle
+
+	// border
+	borderStyleLeft   CellBorderStyle
+	borderColorLeft   *model.PdfColorDeviceRGB
+	borderWidthLeft   float64
+	borderStyleBottom CellBorderStyle
+	borderColorBottom *model.PdfColorDeviceRGB
+	borderWidthBottom float64
+	borderStyleRight  CellBorderStyle
+	borderColorRight  *model.PdfColorDeviceRGB
+	borderWidthRight  float64
+	borderStyleTop    CellBorderStyle
+	borderColorTop    *model.PdfColorDeviceRGB
+	borderWidthTop    float64
+
+	width  *float64
+	height *float64
+}
+
+type htmlBlockStyle struct {
+	TextStyle
+	Bold          bool
+	Italic        bool
+	TextAlignment TextAlignment
+	ElementStyle  *htmlElementStyle
+}
+
+func (style *htmlBlockStyle) getOrCreateElementStyle() *htmlElementStyle {
+	if style.ElementStyle == nil {
+		style.ElementStyle = &htmlElementStyle{}
+	}
+	return style.ElementStyle
+}
+
+func (style *htmlBlockStyle) addEmbeddedCSS(tag string, csstext string) {
+	ss := css.ParseBlock(csstext)
+	for _, s := range ss {
+		switch s.Property {
+		case "text-align":
+			switch s.Value {
+			case "center":
+				style.TextAlignment = TextAlignmentCenter
+			case "left":
+				style.TextAlignment = TextAlignmentLeft
+			case "right":
+				style.TextAlignment = TextAlignmentRight
+			case "justify":
+				style.TextAlignment = TextAlignmentJustify
+			}
+		case "border-width":
+			es := style.getOrCreateElementStyle()
+			w, _ := strconv.ParseFloat(s.Value, 64)
+			es.borderWidthBottom = w
+			es.borderWidthLeft = w
+			es.borderWidthRight = w
+			es.borderWidthTop = w
+		case "border-color":
+			es := style.getOrCreateElementStyle()
+			c := getRGBColorFromHTML(s.Value)
+			es.borderColorBottom = model.NewPdfColorDeviceRGB(c.ToRGB())
+			es.borderColorLeft = es.borderColorBottom
+			es.borderColorRight = es.borderColorBottom
+			es.borderColorTop = es.borderColorBottom
+		case "border-style":
+			es := style.getOrCreateElementStyle()
+			switch s.Value {
+			case "solid":
+				es.borderLineStyle = draw.LineStyleSolid
```

Comment:
Maybe create a method to set the style and line style of the border as to not repeat the code in all switch cases.